src/main/java/gillycarpetaddons/mixins/SpectatorPortalFix.java

```diff
@@ -0,0 +1,55 @@
+package gillycarpetaddons.mixins;
+
+import com.mojang.authlib.GameProfile;
+import gillycarpetaddons.GillyCarpetAddonsSettings;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.MovementType;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+@Mixin(ServerPlayerEntity.class)
+public abstract class SpectatorPortalFix extends PlayerEntity {
```

Comment:
Mixin name should reflect the class it is changing. So it should be something like
`ServerPlayerEntity_SpectatorPlayersUsePortalsMixin` if we use the rule name suggestion above.